### PR TITLE
Fixed the 3D viewer, added the 'save all plots' option to the 2D viewer

### PR DESCRIPTION
**Pull request by thisisrohan**
Debugged and updated the 3D viewer, added the 'save all plots' option to the 2D viewer.

### CODE REVIEW

**Comment by prabhuramachandran** [comment]
Also, what version of ipyvolume does this work with?

File: pysph/tools/ipy_viewer.py [hunk 1]
```diff
@@ -349,7 +358,7 @@ def _configure_plot(self):
         '''
         Set attributes for plotting.
         '''
-
+        plt.ion()
```
[the remark this answers is not in the document]
Comment (thisisrohan) [reply pysph/tools/ipy_viewer.py]: So ipyvolume does not have an inherent legend attribute like matplotlib, and to display the legend for the `3DViewer`, we're plotting it separately in a matplotlib 2D figure that is then being displayed alongside the 3D plot. To get a static plot for the legned, `plt.ioff()` is being used. I thus included `plt.ion()` in  `2DViewer` to make sure we get an interactive figure when the `interactive_plot()` method is called on an instance of  `2DViewer`.
The use case is someone calling `interactive_plot()` on a `2DViewer` instance after calling it on a `3DViewer` instance.

**Comment by thisisrohan** [comment]
This works on ipyvolume 0.5.1

File: pysph/tools/ipy_viewer.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-from pysph.solver.utils import load, get_files
-from IPython.display import display, Image
+import os
+from pysph.solver.utils import load, get_files, mkdir
+from IPython.display import display, Image, clear_output
 import ipywidgets as widgets
 import matplotlib.pyplot as plt
 import matplotlib as mpl
 import numpy as np
+import ipyvolume.pylab as p3
```
[the remark this answers is not in the document]
Comment (thisisrohan) [reply pysph/tools/ipy_viewer.py]: Updated the PR, now it only gets imported if someone calls the `interactive_plot()` method on a `3DViewer` instance.

File: pysph/tools/ipy_viewer.py [hunk 3]
```diff
@@ -656,8 +676,26 @@ def _save_figure_handler(self, change):
...
+        if change['new']:
+            self._widgets.save_figure.disabled = True
+        elif not change['new']:
+            self._widgets.save_figure.disabled = False
+
+    def _save_plot(self):
+
+        abs_path = os.path.abspath(self.path)
+        if self._widgets.save_all_plots.value:
+            mkdir('/all_plots')
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: This should really be `mkdir('all_plots')`!

**Comment by prabhuramachandran** [comment]
I am not too fond of the import in the plot function currently but this does look pretty good, except for one tiny error I spotted, please fix that, I've left a comment.

File: pysph/tools/ipy_viewer.py [hunk 4]
```diff
@@ -656,8 +676,26 @@ def _save_figure_handler(self, change):
...
+            self.figure.savefig(
+                'frame_%s.png' % self._widgets.frame.value,
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: This should be `all_plots/frame...` right?

File: pysph/tools/ipy_viewer.py [hunk 5]
```diff
@@ -683,9 +721,9 @@ def __init__(self, particlearray):
         self.scalar_cmap.owner = self.array_name
         self.velocity_vectors = widgets.Checkbox(
             value=False,
-            description="Vectors",
+            description="Velocity Vectors",
             disabled=False,
-            layout=widgets.Layout(width='100px', display='flex')
+            layout=widgets.Layout(width='300px', display='flex')
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: Do we need fixed sizes here?  Does it render OK on a smaller window?

Comment (thisisrohan) [reply pysph/tools/ipy_viewer.py]: This is just to make sure that the label 'velocity vectors' is displayed completely. It renders fine on a smaller screen, ipywidgets just creates a scroll bar to traverse all the particle arrays.

File: pysph/tools/ipy_viewer.py (hunk 4 above)
Comment (thisisrohan) [reply pysph/tools/ipy_viewer.py]: I've updated this method in the new commit, it no longer uses this function, and yes, it was supposed to be `all_plots/frame..`.

File: pysph/tools/ipy_viewer.py (hunk 3 above)
Comment (thisisrohan) [reply pysph/tools/ipy_viewer.py]: Corrected this in the new commit.

**Comment by prabhuramachandran** [comment]
Thanks! Merging for now.

**Pull request merged**